### PR TITLE
Remove ModuleToGroup from DeploymentConfig

### PR DESCRIPTION
**Pull request by mr0re1**
### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [ ] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -540,13 +540,6 @@ func (s *MySuite) TestHasKind(c *C) {
 
 }
 
-func (s *MySuite) TestCheckModuleAndGroupNames(c *C) {
```
Comment (tpdownes) [inline pkg/config/config_test.go:543]: The overall unit test coverage goes up by 0.1% but I think we are missing some important edge cases in this particular function.

After running `go tool cover -html=coverage.out hpc-toolkit/pkg/config` and `go tool cover -html=coverage.out` I see these fields:

```
                        if _, ok := seen[mod.ID]; ok {
                                return fmt.Errorf("%s: %s used more than once", errorMessages["duplicateID"], mod.ID)
                        }
```

and

```
                        } else if grp.Kind != mod.Kind {
                                return fmt.Errorf(
                                        "%s: deployment group %s, got: %s, wanted: %s",
                                        errorMessages["mixedModule"],
                                        grp.Name, grp.Kind, mod.Kind)
                        }
```

File: pkg/config/config.go [hunk 2]
```diff
@@ -471,47 +469,55 @@ func validateGroupName(name string, usedNames map[string]bool) {
...
 					errorMessages["mixedModule"],
 					grp.Name, grp.Kind, mod.Kind)
 			}
 		}
 	}
-	return moduleToGroup, nil
+	return nil
+}
+
+func modToGrp(groups []DeploymentGroup, modID string) (int, error) {
```
Comment (tpdownes) [inline pkg/config/config.go:498]: I see this function fully covered by unit tests, probably indirectly. Do you feel that's sufficient?

File: pkg/config/config.go [hunk 3]
```diff
@@ -471,47 +469,55 @@ func validateGroupName(name string, usedNames map[string]bool) {
...
+	for iGrp, grp := range groups {
+		for _, mod := range grp.Modules {
```
Comment (tpdownes) [inline pkg/config/config.go:500]: Should we consider [slices.ContainsFunc](https://pkg.go.dev/golang.org/x/exp/slices#ContainsFunc)? An example is already in pkg/config/expand.go. There is probably a marginal cost in performance, but one less thing.

File: pkg/config/config.go [hunk 4]
```diff
@@ -471,47 +469,55 @@ func validateGroupName(name string, usedNames map[string]bool) {
...
+			if mod.ID == modID {
+				return iGrp, nil
+			}
+		}
+	}
+	return 0, fmt.Errorf("Module %s was not found", modID)
```
Comment (tpdownes) [inline pkg/config/config.go:506]: The behavior of [slices.Index](https://pkg.go.dev/golang.org/x/exp/slices#Index) is to return `-1` when it is not found. I think this is probably good behavior to imitate.

File: pkg/config/config_test.go (hunk 1 above)
Comment (tpdownes) [reply pkg/config/config_test.go:543]: This was probably true before, but I'd rather improve this along the way while we are here.

**Review by tpdownes — CHANGES_REQUESTED** [review]
This is a significant improvement by replacing an structure that is passed around from func to func with a single, clear source. The performance penalty is unlikely to be significant and the use of a func simplifies future re-factoring.

There are a few changes I'd like to see or discuss, but this is nearly good to go.

File: pkg/config/config.go (hunk 4 above)
Comment (mr0re1) [reply pkg/config/config.go:506]: Thank you! I did forget that Go doesn't support negative indexes )

File: pkg/config/config.go (hunk 2 above)
Comment (mr0re1) [reply pkg/config/config.go:498]: Quite the opposite, I'm disturbed by abundance of tests for "private" functions )

File: pkg/config/config_test.go (hunk 1 above)
Comment (mr0re1) [reply pkg/config/config_test.go:543]: Added test for ` used more than once `. But unearthed other problems in this function, fixing + adding test is going out of scope of this PR.

**Review by tpdownes — APPROVED** [review]
LGTM. If you believe that a useful test is missing (mismatched Kind?), please create a task for it in the backlog.
